src/System.Private.CoreLib/src/System/Diagnostics/Eventing/RuntimeEventSource.cs

```diff
@@ -0,0 +1,143 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+
+using System.Runtime.InteropServices;
+using Microsoft.Win32;
+using Microsoft.Win32.SafeHandles;
+
+using System.Runtime.InteropServices.ComTypes;
+using System.StubHelpers;
+
+
+namespace System.Diagnostics.Tracing
+{
+    /// <summary>
+    /// RuntimeEventSource is an EventSource that represents events emitted by the managed runtime.
+    /// </summary>
+    [EventSource(Name = "System.Runtime")]
+    internal sealed class RuntimeEventSource : EventSource
+    {
+        // Process-level EventCounters
+        private EventCounter _totalProcessTimeCounter;
```

Comment:
Would it make more sense to have a `List<EventCounter>` and add all of them on constructor?

Reply:
If I use a List then I need to keep track of which counter is stored in which index of the EventCounter. Each counter (in the future) might get polled at different rate, etc., so I didn't want to put them in a list. I guess I could do that and keep some kind of ```enum``` to track that, but I  didn't think that that was clean. But if that seems better then it's not a big deal to change it so let me know what you think!

Reply:
Or just have an array of them. Like:

```
new EventCounter[] {
     new EventCounter(...),
     new EventCounter(...),
};
```

Disposing all of them can be a simple loop once you do that.

Reply:
> Each counter (in the future) might get polled at different rate

You can refactor that in future if necessary. Future proofing rarely works well.